server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponse.java

```diff
@@ -81,8 +81,8 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeMap(responseMap, StreamOutput::writeString, (valueOut, fc) -> fc.writeTo(valueOut));
         }
 
-        List<FieldCapabilitiesIndexResponse> getResponses() {
-            return indices.stream().map(index -> new FieldCapabilitiesIndexResponse(index, indexMappingHash, responseMap, true)).toList();
+        Stream<FieldCapabilitiesIndexResponse> getResponses() {
```

Comment:
No need to collect this if we turn it into a stream in the only callsite right away anyway.